### DISCUSSION TITLE
User shared folder download features.

### DISCUSSION THREAD

**Discussion by rashmimaharana** [drakkan/sftpgo#677] [Q&A]
Hi @drakkan 

Is there any way to download shared folder/files from user using the generated link. I could see the option to upload files and that works fine. Thanks for this feature :) 

Scenario :
1> Created user account.
2> Created a directory called "shared" using client GUI.
3> Shared that directory with write permission.
4> Uploaded file to shared directory using generated link ( curl -F filenames=@SFTPGO.doc http://10.125.XX.XXX:8080/web/client/pubshares/cdyXzdmf... )
5> Would like to download.

![image](https://user-images.githubusercontent.com/10846216/149339007-df45193d-2e2e-43b8-b570-cca7b376df66.png)

**Comment by drakkan** [comment]
Hi,

not sure to fully understand the question. If you create a new read only share and set the path to `/shared` you should be able to download the `/shared` directory as zip. Do you expect something different?

**Reply by rashmimaharana** [reply]
Sorry for some missing info. Here is what i am looking for.
Here in screen shot where user "max" is sharing a directory called "shared" with write permission. I would like to download the same using rest api but without any token. The screenshot given above is describing how to upload the files to shared folder using "share access link".
<img width="959" alt="Capture12" src="https://user-images.githubusercontent.com/10846216/149348861-c0cbbe28-abaf-4f30-adc7-efebc57cd302.PNG">

**Reply by drakkan** [reply]
Sorry, today is really a crazy day for me and I could miss something. If you create a new share called for example `shared read` with `Read` scope and set the same path as the `shared` share in the above screenshot, you should get a link that allows to download the whole shared content as single zip file (and you can disable authentication, max tokens etc.), do you want to download single files one by one instead?

**Reply by rashmimaharana** [reply]
Thanks @drakkan for the info.
Yes, any steps to download files as uncompressed. Also download single file as well.

**Reply by drakkan** [reply]
> Thanks @drakkan for the info. Yes, any steps to download files as uncompressed. Also download single file as well.

I'll try to make the shares with read scope "browsable" in the future. I am busy with several requests from people sponsoring the project now, this feature was already on my private TODO but it may take a while

**Reply by drakkan** [reply]
> Thanks @drakkan for the info. Yes, any steps to download files as uncompressed. Also download single file as well.

see this commit https://github.com/drakkan/sftpgo/commit/9382db751ca227cdd8e8ef7d01e62beb3642bba0
